subarray.go

```diff
@@ -52,6 +52,101 @@ func (sa *Subarray) Free() {
 	}
 }
 
+// SetSubarray sets a subarray, defined in the order dimensions were added.
```

Comment:
Should be `SetSubArray` instead of `SetSubarray`, otherwise Go's doc tool may not pick up on this.